kubernetes/resource_kubernetes_persistent_volume_test.go

```diff
@@ -1313,6 +1357,99 @@ resource "azurerm_storage_container" "test" {
 `, name, location)
 }
 
+func testAccKubernetesPersistentVolumeConfig_azure_PersistentVolumeAzureFile(name, secretName string) string {
+	return fmt.Sprintf(`resource "kubernetes_persistent_volume" "test" {
+  metadata {
+    name = %[1]q
+  }
+  spec {
+    capacity = {
+      storage = "1Gi"
+    }
+    access_modes = ["ReadWriteOnce"]
+    persistent_volume_source {
+      azure_file {
+        secret_name      = %[2]q
+        share_name       = %[1]q
+        read_only        = false
+      }
+    }
+  }
+}`, name, secretName)
+}
+
+func testAccKubernetesPersistentVolumeConfig_azure_PersistentVolumeAzureFileNamespace(name, namespace, secretName string) string {
+	return fmt.Sprintf(`resource "kubernetes_namespace" "test" {
+  metadata {
+    name = %[2]q
+  }
+}
+
+resource "kubernetes_secret" "test" {
+  metadata {
+    name      = %[3]q
+    namespace = %[2]q
+  }
+
+  data = {
+    azurestorageaccountname = azurerm_storage_account.test.name
+    azurestorageaccountkey  = azurerm_storage_account.test.primary_access_key
+  }
+}
```

Comment:
These tests look great! I'm especially excited to see an example demonstrating a feature I haven't used before. Thanks for this!

Reply:
Ideally we would also create a pod and validate that the volume is actually mounted but this is better than nothing :)

Reply:
Good idea! I ended up doing that manually to test. I'll paste it here in case anyone is randomly googling how to mount azure file shares to their pod.
 
https://gist.githubusercontent.com/dak1n1/43a199ce6b066d894f5a878f3f588561/raw/d5f86fa57f4dd79bda4d93c1364897a180b9b8a1/gistfile1.txt